### DISCUSSION TITLE
16.18

### DISCUSSION THREAD

**Discussion by RoboMWM** [GriefPrevention/GriefPrevention#1821] [General]
No use in waiting to release if you don't release at all.

Lots of changes, and this auto-generate release notes really is handy. I considered using a bot, or making my own changelog, but in the end, good commit messages will just go hand-in-hand with this. Given that most contributions are PRs, my own contributions will probably be PRs as well, and thus should have a cleaner commit history in the first place.

Unless there are any severe bugs with this version (should be a discussion attached to this release), this will also be uploaded to BukkitDev shortly after.

Should be compatible with Minecraft 1.17 and 1.18.

## What's Changed
* Check before casting to mob by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1281
* Fix ignoring height by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1260
* Prevent nametag usage unless player can kill target by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1337
* Include scope in dependency information by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1390
* Change method signature of allowBreak method in main class by @zhenghanlee in https://github.com/TechFortress/GriefPrevention/pull/1406
* Treat glow item frames like item frames by @drbot7 in https://github.com/TechFortress/GriefPrevention/pull/1413
* Add ClaimPermissionCheckEvent by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1006
* Fix deprecated claim API by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1439
* Update to 1.17 by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1448
* Set up GitHub actions maven workflow by @nouish in https://github.com/TechFortress/GriefPrevention/pull/1454
* Actions workflow: use the right default branch by @nouish in https://github.com/TechFortress/GriefPrevention/pull/1456
* protect glow berries by @drbot7 in https://github.com/TechFortress/GriefPrevention/pull/1459
* Fix falling block and vehicle block change handling by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1424
* Protect Pumpkin Carving by @drbot7 in https://github.com/TechFortress/GriefPrevention/pull/1473
* Option to unsilence bans by @drbot7 in https://github.com/TechFortress/GriefPrevention/pull/1480
* Fix NPE when using `allowBuild` in a claim that the player has permission to build by @MrPowerGamerBR in https://github.com/TechFortress/GriefPrevention/pull/1500
* Fix max depth below world height by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1502
* Added (glow) ink sac to materials list for dyes by @multirp in https://github.com/TechFortress/GriefPrevention/pull/1508
* Protect candles from being extinguished by @drbot7 in https://github.com/TechFortress/GriefPrevention/pull/1526
* resolve NPE when calling Claim.allowEdit by @loganb1max in https://github.com/TechFortress/GriefPrevention/pull/1580
* Fix check for expiring chest claims by @Bobcat00 in https://github.com/TechFortress/GriefPrevention/pull/1600
* Fix Non-Player Splash Potions by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1636
* Convert to YAML bug report template by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1638
* Update Java and fix AppVeyor by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1663
* ClaimTransferEvent by @bertek41 in https://github.com/TechFortress/GriefPrevention/pull/1641
* Sanitize claim depth on extend/resize by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1598
* Protect rooted_dirt by @drbot7 in https://github.com/TechFortress/GriefPrevention/pull/1541
* Better handling of usernames with periods by @Camotoy in https://github.com/TechFortress/GriefPrevention/pull/1543
* Don't fire EntityDamageByEntityEvent when renaming by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1680
* Correct WorldGuard integration by @Kamilkime in https://github.com/TechFortress/GriefPrevention/pull/1630
* Fix 1.18 biome issues by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1694
* Clean up events for add-ons by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1706
* Fix new depth not being set correctly by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1736
* Rework Visualizations by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1741
* Fix incorrect permission bypass for Edit ClaimPermission by @Jikoo in https://github.com/TechFortress/GriefPrevention/pull/1772

## New Contributors
* @zhenghanlee made their first contribution in https://github.com/TechFortress/GriefPrevention/pull/1406
* @MrPowerGamerBR made their first contribution in https://github.com/TechFortress/GriefPrevention/pull/1500
* @multirp made their first contribution in https://github.com/TechFortress/GriefPrevention/pull/1508
* @loganb1max made their first contribution in https://github.com/TechFortress/GriefPrevention/pull/1580
* @bertek41 made their first contribution in https://github.com/TechFortress/GriefPrevention/pull/1641
* @Camotoy made their first contribution in https://github.com/TechFortress/GriefPrevention/pull/1543
* @Kamilkime made their first contribution in https://github.com/TechFortress/GriefPrevention/pull/1630

**Full Changelog**: https://github.com/TechFortress/GriefPrevention/compare/16.17.1...16.18

<hr /><em>This discussion was created from the release <a href='https://github.com/TechFortress/GriefPrevention/releases/tag/16.18'>16.18</a>.</em>

**Comment by zNinjaG0DHG** [comment]
Hi, i can't protect under the -0y the protection only affect to 0-319y

**Reply by Jikoo** [reply]
Ensure that you have actually edited your config's `MaximumDepth` to allow claims to extend below 0. New value should likely be -64 for most people.

/e: MaximumDepth, maxDepth is the internal reference.

**Reply by zNinjaG0DHG** [reply]
Hi, i don't have this option on my config. I have deleted my whole config to generate a new file. But i haven't this "MaxDepth"

**Reply by Jikoo** [reply]
The full path is `GriefPrevention.Claims.MaximumDepth`. If you deleted your config, it will be set to the default value, which is sufficiently low as of 16.18.

/e: MaximumDepth, maxDepth is the internal reference.

**Reply by zNinjaG0DHG** [reply]
GriefPrevention:
  SeaLevelOverrides:
    spawn: -1
    spawn_the_end: -1
    survival_nether: -1
    evento: -1
    survival: -1
    cultivos: -1
    comida: -1
    survival_the_end: -1
    tutorial: -1
    mina: -1
  Claims:
    Mode:
      survival_nether: Survival
      tutorial: Survival
      mina: Survival
      comida: Survival
      spawn: Survival
      survival_the_end: Survival
      evento: Survival
      cultivos: Survival
      survival: Survival
      spawn_the_end: Disabled
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 100
    Claim Blocks Accrued Per Hour:
      Default: 100
    Max Accrued Claim Blocks:
      Default: 80000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: -2147483648
    InvestigationTool: STICK
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 7
      UnusedClaimDays: 14
      AllClaims:
        DaysInactive: 60
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true

**Reply by zNinjaG0DHG** [reply]
You can see this option don't appears by default

**Reply by bigpresh** [reply]
It's [GriefPrevention.Claims.MaximumDepth](https://github.com/TechFortress/GriefPrevention/blob/master/src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java#L561) (not `MaxDepth`) which does appear in your config with a suitably deep value.

Can you confirm that claims aren't extending below y=0 even if you mine down / place blocks below them, which should cause the claim to auto-extend downwards?

**Reply by zNinjaG0DHG** [reply]
Yes, with this config i can mine/placeblock below them. I can place blocks below 52 exactly

**Reply by DeSinc** [reply]
> The full path is `GriefPrevention.Claims.MaxDepth`. If you deleted your config, it will be set to the default value, which is sufficiently low as of 16.18.

This didn't work when I added a new line called 'MaxDepth' to the config. I generated a new config to figure out why and the value that needs to change is MaximumDepth. Posting here so others can see

EDIT: what?? a whole bunch of replies just appeared when I posted this reply lol. I didn't even see any of these other replies here before.
(maybe if you could edit your original comment that says incorrectly to change MaxDepth to 'MaximumDepth' instead so it doesn't catch people like me coming here to find the solution)

**Reply by Jikoo** [reply]
Presumably you hadn't expanded the rest of the discussion. Good point though 👍
